### PR TITLE
test: add jit, grad, correctness, and edge case tests for Normal

### PR DESCRIPTION
**Pull request by professor314**

### CODE REVIEW

File: tests/normal_test.py [hunk 1]
```diff
@@ -220,3 +220,169 @@ def summed_dist(loc, scale):
 
         x = jnp.array([[[1]], [[2]], [[3]]])
         np.testing.assert_allclose(actual.log_prob(x), expected.log_prob(x), rtol=1e-6)
+
+    # ===== Compatibility: jit =====
```
Comment (lockwo) [inline tests/normal_test.py:224]: un-needed comment + tests are failing
